File: BST/bst-height-hr.js
/**
 * Time O(n)
 * Space O(1)
 */
function getHeight(root) {
    return  getBstHeightHelper(root) - 1;
}
    
function getBstHeightHelper(rootNode){
    // Write your code here.
    if(rootNode == null){
        return 0;
    }; 
    const lftHeight = getBstHeightHelper(rootNode.left);
    const rightHeight = getBstHeightHelper(rootNode.right);
    return Math.max(lftHeight, rightHeight) + 1;
}

File: BST/bst-height-hr.java
	public static int height(Node root) {
        // Write your code here.
        // remove  1 level for the root node considered 0 
        return  heightHelper(root) - 1;
    }
    // performs a post order traversal and returns the max of curr left and right  + 1  
    private static int heightHelper(Node root) {
        // Write your code here.
        if(root == null){
            return 0;
        }; 
        int lftHeight = heightHelper(root.left);
        int rightHeight = heightHelper(root.right);
        return Math.max(lftHeight, rightHeight) + 1;
    }
    